File: my-blog-backend/src/main/java/com/Chirpchump/my_blog_backend/service/CommentService.java
package com.Chirpchump.my_blog_backend.service;

import com.Chirpchump.my_blog_backend.dto.CommentRequest;
import com.Chirpchump.my_blog_backend.dto.CommentResponse;
import java.util.List;
import java.util.Map;

public interface CommentService {

    /**
     * 根据文章ID获取树状评论列表
     * @param postId 文章ID
     * @return 评论树列表
     */
    List<CommentResponse> getCommentsByPostId(Long postId);

    /**
     * 创建一条新评论
     * @param commentRequest 评论请求数据
     * @return 创建后的评论
     */
    CommentResponse createComment(CommentRequest commentRequest);

    /**
     * 删除一条评论 (包含权限校验)
     * @param commentId 评论ID
     */
    void deleteComment(Long commentId);

    /**
     * 点赞一条评论
     * @param commentId 评论ID
     * @return 包含新点赞数和点赞状态的Map
     */
    Map<String, Object> likeComment(Long commentId);

    /**
     * 取消点赞一条评论
     * @param commentId 评论ID
     * @return 包含新点赞数和点赞状态的Map
     */
    Map<String, Object> unlikeComment(Long commentId);
}

File: my-blog-frontend/src/api/commentService.js
import apiClient from './axiosInstance';

/**
 * 根据文章ID获取评论列表
 * @param {number} postId
 * @returns {Promise<any>}
 */
export const fetchCommentsByPost = (postId) => {
    return apiClient.get(`/comments/post/${postId}`);
};

/**
 * 创建新评论
 * @param {object} data - { content, postId, parentId }
 * @returns {Promise<any>}
 */
export const createComment = (data) => {
    return apiClient.post('/comments', data);
};

/**
 * 删除评论
 * @param {number} commentId
 * @returns {Promise<any>}
 */
export const deleteComment = (commentId) => {
    return apiClient.delete(`/comments/${commentId}`);
};

/**
 * 点赞评论
 * @param {number} commentId
 * @returns {Promise<any>}
 */
export const likeComment = (commentId) => {
    return apiClient.post(`/comments/${commentId}/like`);
};

/**
 * 取消点赞评论
 * @param {number} commentId
 * @returns {Promise<any>}
 */
export const unlikeComment = (commentId) => {
    return apiClient.delete(`/comments/${commentId}/like`);
};